### ISSUE TITLE
Type reference to require import resolves to static side

### ISSUE THREAD

**Issue by sandersn** [microsoft/TypeScript#34802]
```ts
// @filename: ex.d.ts
export class C {
  start: number
  end: number
}
// @filename: test.js
const C = require('./ex').C
const C = require('./ex').C;
/** @type {C} c */
var c = new C()
```

Expected: No error, and completions for `c.start` and `c.end`

Actual: Error, "Property 'prototype' is missing in type 'C' but required in type 'typeof C'."

**Comment by sandersn** [comment]
The new syntactic check for `require` calls in getTypeFromJSDocValueReference looks for exactly `= require(STRING)`. It's easy to fix, but this makes me wonder whether the syntactic approach is the right one.

**Comment by sandersn** [comment]
On the other hand, the check I added is equivalent to `import("x").a.b`, which only has two forms, typeof being the other, so it's likely that this check is good enough.
